### PR TITLE
bottom: init at 0.4.7

### PR DESCRIPTION
**Pull request by berbiche**
##### Motivation for this change

bottom is "a cross-platform graphical process/system monitor with a customizable interface and a multitude of features".
[ytop](https://github.com/cjbassi/ytop) has been archived by its owner.
`bottom` has been recommended by `ytop`'s creator.

Some tests are failing because they call the binary and it tries to create its default file in `$XDG_CONFIG_HOME/bottom/bottom.toml`.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by berbiche** [comment]
ccing `ytop`'s maintainer: @sikmir

**Review by r-burns — CHANGES_REQUESTED** [review]
Works great for me on non-NixOS linux.

Needs IOKit to compile on macOS, and the resulting executable just shows empty graphs. Is that a bug, or is this linux-only?

File: pkgs/tools/system/bottom/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{ stdenv, fetchFromGitHub, rustPlatform }:
```
[the remark this answers is not in the document]
Comment (berbiche) [reply pkgs/tools/system/bottom/default.nix]: Thanks, fixed!

File: pkgs/tools/system/bottom/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  src = fetchFromGitHub {
+    owner = "ClementTsang";
+    repo = pname;
+    rev = version;
+    sha256 = "rDcJ5XF7L13MKZ8/J4sYD+UqC+HkZvxRtDkY9IVLH50=";
+  };
+
+  cargoSha256 = "XeX6QM0a628mcaptNZkKAvDnGfW5tx+aWNBpMyjz44M=";
+
```
[the remark this answers is not in the document]
Comment (berbiche) [reply pkgs/tools/system/bottom/default.nix:17]: Thanks, fixed!

**Comment by berbiche** [comment]
I tried building this on a Darwin machine but I had SSL certificate errors when fetching dependencies with Cargo so I'll leave the testing to @r-burns

**Review by r-burns — APPROVED** [review]
Awesome, thanks for taking care of that, compiles + runs for me now. I see that bottom claims to support macOS so I've filed an issue upstream. As for this PR, LGTM 🎉

**Review by stigtsp — APPROVED** [review]
LGTM

Built and tested OK on `linux x64`

Result of `nixpkgs-review pr 99126` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>bottom</li>
  </ul>
</details>

**Pull request merged**